server/sonar-server/src/main/java/org/sonar/server/component/db/ComponentDao.java

```diff
@@ -182,18 +181,36 @@ protected void doDeleteByKey(DbSession session, String key) {
   public List<ComponentDto> selectProvisionedProjects(DbSession session, SearchOptions searchOptions, @Nullable String query) {
     Map<String, String> parameters = new HashMap<>();
     parameters.put("qualifier", Qualifiers.PROJECT);
-    if (query != null) {
-      parameters.put("query", "%" + query + "%");
-    }
+    parameters.put("query", toSqlPartialSearchQuery(query));
+
     return mapper(session).selectProvisionedProjects(parameters, new RowBounds(searchOptions.getOffset(), searchOptions.getLimit()));
   }
 
   public int countProvisionedProjects(DbSession session, @Nullable String query) {
     Map<String, String> parameters = new HashMap<>();
     parameters.put("qualifier", Qualifiers.PROJECT);
-    if (query != null) {
-      parameters.put("query", "%" + query + "%");
-    }
+    parameters.put("query", toSqlPartialSearchQuery(query));
+
     return mapper(session).countProvisionedProjects(parameters);
   }
+
+  public List<ComponentDto> selectGhostProjects(DbSession session, @Nullable String query, SearchOptions options) {
+    Map<String, String> parameters = new HashMap<>();
+    parameters.put("qualifier", Qualifiers.PROJECT);
+    parameters.put("query", toSqlPartialSearchQuery(query));
+
+    return mapper(session).selectGhostProjects(parameters, new RowBounds(options.getOffset(), options.getLimit()));
+  }
+
+  public long countGhostProjects(DbSession session, @Nullable String query) {
+    Map<String, String> parameters = new HashMap<>();
+    parameters.put("qualifier", Qualifiers.PROJECT);
+    parameters.put("query", toSqlPartialSearchQuery(query));
+
+    return mapper(session).countGhostProjects(parameters);
+  }
+
+  private String toSqlPartialSearchQuery(@Nullable String query) {
+    return query == null ? "%" : "%" + query.toUpperCase() + "%";
```

Comment:
I try to avoid dynamic SQL to just have SQL in the Mapper.xml class. It's mostly a style issue. I guess I prefer to have several ComponentMapper classes, but it's possible to have quickly a lot of classes